src/crawler/iCrawler.py

```diff
@@ -32,14 +32,5 @@ async def check_new_url(self, url: str) -> bool:
         urls = await self.database.fetch_video_url_list()
         return url not in urls
 
-
-    #helper method to publish item to database
-    async def publish_metadata(self, item:MetaDataItem):
-        if self.database is None:
-            raise UndefinedDatabaseException()
-
-        await self.database.publish_new_metadata(item)
-
     def run(self, obj : Dict[str, Any]):
-        metadata_item = asyncio.run(self.next_downloadable())
-        asyncio.run(self.publish_metadata(metadata_item))
+        return asyncio.run(self.next_downloadable())
```

Comment:
is there a reason we use asyncio here? Since the crawlers are iExecutors they are already running on a separate process

Reply:
I believe its necessary since the inherited run function is not async, so its needed so it can call next_downloadable. If we decide that run should be async for all Executors (which would make sense, then this needs to be changed.